crates/hyperdrive-math/src/lp.rs

```diff
@@ -353,6 +412,122 @@ impl State {
             ))
             .unwrap()
     }
+
+    /// Calculates the number of share reserves that are not reserved by open
+    /// positions.
+    fn calculate_idle_share_reserves(&self) -> FixedPoint {
+        let long_exposure = self.long_exposure().div_up(self.vault_share_price());
+        let idle_shares = {
+            if self.share_reserves() > long_exposure + self.minimum_share_reserves() {
+                return self.share_reserves() - long_exposure - self.minimum_share_reserves();
+            }
+            fixed!(0)
+        };
+
+        idle_shares
+    }
+
+    /// Given a signed bond amount, this function calculates the negation of the
+    /// derivative of `calculateSharesOutGivenBondsIn` when the bond amount is
+    /// positive or the derivative of `calculateSharesInGivenBondsOut` when the
+    /// bond amount is negative.
+    /// Note that the state is the present state of the pool and original values
+    /// passed in as parameters.  Present sate variables are not expressly
+    /// paased in because so that downstream function like kUp() can still be
+    /// used.
+    fn calculate_shares_delta_given_bonds_delta_derivative(
+        &self,
+        bond_amount: I256,
+        original_share_reserves: FixedPoint,
+        original_bond_reserves: FixedPoint,
+        original_effective_share_reserves: FixedPoint,
+        original_share_adjustment: I256,
+    ) -> Result<FixedPoint> {
+        // Calculate the bond reserves after the bond amount is applied.
+        let bond_reserves_after = if bond_amount >= I256::zero() {
+            self.bond_reserves() + bond_amount.into()
+        } else {
+            let bond_amount = FixedPoint::from(U256::try_from(-bond_amount)?);
+            if bond_amount < self.bond_reserves() {
+                self.bond_reserves() - bond_amount
+            } else {
+                return Err(eyre!("Calculating the bond reserves underflows"));
+            }
+        };
+
+        // NOTE: Round up since this is on the rhs of the final subtraction.
+        //
+        // derivative = c * (mu * z_e(x)) ** -t_s +
+        //              (y / z_e) * (y(x)) ** -t_s -
+        //              (y / z_e) * (y(x) + dy) ** -t_s
+        let effective_share_reserves = self.effective_share_reserves();
+        let derivative = self.vault_share_price().div_up(
+            self.initial_vault_share_price()
+                .mul_down(effective_share_reserves)
+                .pow(self.time_stretch()),
+        ) + original_bond_reserves.div_up(
+            original_effective_share_reserves
+                .mul_down(self.bond_reserves().pow(self.time_stretch())),
+        );
+
+        // NOTE: Rounding this down rounds the subtraction up.
+        let right_hand_side = original_bond_reserves.div_down(
```

Comment:
Let's just use the same naming as in the contracts, unless there is something that I'm missing. Switching the style is one thing, but I think it'll be easier to follow if it's exactly the same.